network-config-analyzer/GenericYamlParser.py

```diff
@@ -118,6 +132,38 @@ def check_fields_validity(self, dict_to_check, dict_name, allowed_keys, allowed_
             self.syntax_error(f'{bad_policy_keys.pop()} is not a valid entry in the specification of {dict_name}',
                               dict_to_check)
 
+    def parse_generic_yaml_objects_fields(self, yaml_object, object_kind, object_version, layer_keywords, spec_required=False):
+        """
+        Parse and check validity of the common fields in network yaml objects as policies and services.
+        e.g. kind, apiVersion, metadata and spec
+        :param dict yaml_object: the source dict of the network yaml object to be parsed
+        :param list object_kind : the possible kind value(s) of current object
```

Comment:
When possible, be more specific with param type
```suggestion
        :param list[str] object_kind : the possible kind value(s) of current object
```

Reply:
Other params on this function can be specified with more details on their type